packages/effect/src/DateTime.ts

```diff
@@ -0,0 +1,1991 @@
+ * return the result.
+ *
+ * @since 3.6.0
+ * @category mapping
+ * @example
+ * import { DateTime } from "effect"
+ *
+ * // get the date in milliseconds
+ * DateTime.unsafeMake(0).pipe(
+ *   DateTime.withDateUtc((date) => date.getTime())
+ * )
+ */
+export const withDateUtc: {
+  <A>(f: (date: Date) => A): (self: DateTime) => A
+  <A>(self: DateTime, f: (date: Date) => A): A
+} = dual(2, <A>(self: DateTime, f: (date: Date) => A): A => f(toDateUtc(self)))
+
+/**
+ * @since 3.6.0
+ * @category mapping
+ */
+export const match: {
+  <A, B>(options: {
+    readonly onUtc: (_: Utc) => A
+    readonly onZoned: (_: Zoned) => B
+  }): (self: DateTime) => A | B
+  <A, B>(self: DateTime, options: {
+    readonly onUtc: (_: Utc) => A
+    readonly onZoned: (_: Zoned) => B
+  }): A | B
+} = dual(2, <A, B>(self: DateTime, options: {
+  readonly onUtc: (_: Utc) => A
+  readonly onZoned: (_: Zoned) => B
+}): A | B => self._tag === "Utc" ? options.onUtc(self) : options.onZoned(self))
+
+// =============================================================================
+// math
+// =============================================================================
+
+/**
+ * Add the given `Duration` to a `DateTime`.
+ *
+ * @since 3.6.0
+ * @category math
+ * @example
+ * import { DateTime } from "effect"
+ *
+ * // add 5 minutes
+ * DateTime.unsafeMake(0).pipe(
+ *   DateTime.addDuration("5 minutes")
+ * )
+ */
+export const addDuration: {
+  (duration: Duration.DurationInput): <A extends DateTime>(self: A) => DateTime.PreserveZone<A>
+  <A extends DateTime>(self: A, duration: Duration.DurationInput): DateTime.PreserveZone<A>
+} = dual(
+  2,
+  (self: DateTime, duration: Duration.DurationInput): DateTime =>
+    mapEpochMillis(self, (millis) => millis + Duration.toMillis(duration))
+)
+
+/**
+ * Subtract the given `Duration` from a `DateTime`.
+ *
+ * @since 3.6.0
+ * @category math
+ * @example
+ * import { DateTime } from "effect"
+ *
+ * // subtract 5 minutes
+ * DateTime.unsafeMake(0).pipe(
+ *   DateTime.subtractDuration("5 minutes")
+ * )
+ */
+export const subtractDuration: {
+  (duration: Duration.DurationInput): <A extends DateTime>(self: A) => DateTime.PreserveZone<A>
+  <A extends DateTime>(self: A, duration: Duration.DurationInput): DateTime.PreserveZone<A>
+} = dual(
+  2,
+  (self: DateTime, duration: Duration.DurationInput): DateTime =>
+    mapEpochMillis(self, (millis) => millis - Duration.toMillis(duration))
+)
+
+const addMillis = (date: DateTime, amount: number): DateTime => mapEpochMillis(date, (millis) => millis + amount)
+
+/**
+ * Add the given `amount` of `unit`'s to a `DateTime`.
+ *
+ * The time zone is taken into account when adding days, weeks, months, and
+ * years.
+ *
+ * @since 3.6.0
+ * @category math
+ * @example
+ * import { DateTime } from "effect"
+ *
+ * // add 5 minutes
+ * DateTime.unsafeMake(0).pipe(
+ *   DateTime.add(5, "minutes")
+ * )
+ */
+export const add: {
```

Comment:
For this and `subtract`, would it make sense for it to instead accept a `Partial<DateTime.Parts>` object? That way, users could add various units at once, and you'd be saved from dealing with so many strings. This also aligns with the `Temporal` API.